python/tchannel/thrift/server.py

```diff
@@ -82,23 +87,28 @@ def build_handler(service_module, method_name, handler):
     args_type = getattr(service_module, method_name + '_args')
     result_type = getattr(service_module, method_name + '_result')
 
-    def thrift_handler(request, response, connection):
+    @tornado.gen.coroutine
+    def thrift_handler(request, response, proxy):
         # TODO: Fix arg scheme passing
         # assert request.arg_scheme == 'thrift', (
         #     "Invalid arg scheme %s" % request.arg_scheme
         # )
 
+        body = yield request.get_body()
         args = args_type()
         args.read(
             TBinaryProtocol.TBinaryProtocolAccelerated(
-                TTransport.TMemoryBuffer(request.body)
+                TTransport.TMemoryBuffer(body)
             )
         )
 
         result = result_type()
         try:
+            thrift_args = [
+                getattr(args, spec[2]) for spec in args.thrift_spec[1:]]
+            thrift_args.append(proxy)
```

Comment:
No. This is a bad way to inject the proxy. An arbitrary "last argument is proxy" is a bad interface and really paints us in a corner if we want to make any change to the scheme. It breaks all existing Thrift handlers (which currently just conform to the interface specified for the service in the IDL).

A possible better way would be to make it an instance attribute and construct a new Handler for each request with the Proxy for that request set to this value. But we wouldn't want to enforce a constructor shape either.

We need to have an offline discussion about the Thrift interface overall before we make a change like this.